### PR TITLE
Ignore cancelled course rounds

### PR DESCRIPTION
**Pull request by me97esn**

### CODE REVIEW

File: packages/t2l-backend/src/externalApis/ladokApi/index.ts [hunk 1]
```diff
@@ -85,10 +85,22 @@ export function getSkaFinnasStudenter(aktivitetstillfalleUID: string) {
...
-    .then((response) => response.body);
+    .then((response) => response.body)
+    .catch((e) => {
+      const errorBody = e.response.body;
+      if (
+        errorBody.Detaljkod === "resultat.fel.detaljkod.utbildningstillfalle" &&
+        errorBody.Felgrupp === "commons.fel.grupp.obligatoriskt_varde_saknas"
+      ) {
+        // This looks like the course round doesn't exist in Ladok. Ignore it.
+        return null;
```
Comment (me97esn) [inline packages/t2l-backend/src/externalApis/ladokApi/index.ts:99]: We should probably log this, it's a bit dangerous to silently skip things

File: packages/t2l-backend/src/apiHandlers/ladokGrades.ts [hunk 2]
```diff
@@ -31,8 +31,8 @@ async function checkUtbildningsinstansInKurstillfalle(
   const ladokKurstillfalle = await getKurstillfalleStructure(kurstillfalleUID);
 
   const isFinalGrade =
-    ladokKurstillfalle.UtbildningsinstansUID === utbildningsinstansUID;
-  const isModule = ladokKurstillfalle.IngaendeMoment.some(
+    ladokKurstillfalle?.UtbildningsinstansUID === utbildningsinstansUID;
```
Comment (exacs) [inline packages/t2l-backend/src/apiHandlers/ladokGrades.ts:34]: I think if the kurstillfälle does not exist in Ladok, the function should throw a more specific error. As it is implemented now, it throws the error in line 39 `Provided utbildningsinstans [${utbildningsinstansUID}] doesn't refer to a module or the final grade of the kurstillfälle [${kurstillfalleUID}]` which is not true.

The function definition (line 26) "Checks if the given `utbildningsinstans` belongs to the given `kurstillfalle`" should also be updated

File: packages/t2l-backend/src/externalApis/ladokApi/index.ts (hunk 1 above)
Comment (exacs) [reply packages/t2l-backend/src/externalApis/ladokApi/index.ts:99]: I am not sure if the function should return `null` in this case or throw a more detailed error, something like `LadokApiError` or something like that, but that can be done in the future.

The function spec (lines 85-86) is updated.

Agree to log things, but definitely _not_ in this module. As described in line 2:

```
* This module are functions to call Ladok API. They do not contain any logic.
```

File: packages/t2l-backend/src/externalApis/ladokApi/types.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ export interface Kurstillfalle {
   UtbildningsinstansUID: string;
   Utbildningskod: string;
   Versionsnummer: number;
+  Installt?: boolean;
```
Comment (exacs) [inline packages/t2l-backend/src/externalApis/ladokApi/types.ts:47]: This is not used and doesn't need to be defined

**Review by exacs — CHANGES_REQUESTED** [review]

**Pull request merged**